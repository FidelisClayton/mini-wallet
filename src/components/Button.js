import React from 'react'
import { css } from 'emotion'

const styles = css({
  height: '40px',
  border: 0,
  borderRadius: '5px',
  width: '100%',
  backgroundColor: '#4CAF50',
  color: 'white',
  fontWeight: 'bold',
  fontSize: '16px'
})

const Button = ({
  children,
  className,
  ...props
}) => (
  <button
    className={`c-button ${styles} ${className}`}
    {...props}
  >
    { children }
  </button>
)

export default Button
