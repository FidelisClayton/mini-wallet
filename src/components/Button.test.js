import React from 'react'
import { shallow } from 'enzyme'

import Button from './Button'

describe('Component: <Button />', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Button>Button</Button>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
