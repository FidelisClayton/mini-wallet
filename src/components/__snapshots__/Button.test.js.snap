// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Button /> should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    Button
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Button",
      "className": "c-button css-dckfuz undefined",
    },
    "ref": null,
    "rendered": "Button",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Button",
        "className": "c-button css-dckfuz undefined",
      },
      "ref": null,
      "rendered": "Button",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
